### ISSUE TITLE
QA release: v3.11.0-rc1 - Syscheck

### ISSUE THREAD

**Issue by chemamartinez** [wazuh/wazuh-qa#193]
# Testing: Syscheck

| Version | Revision | Branch |
| --- | --- | --- |
| 3.11.0 | rc1 | 3.11 |

## Any

- [x] Check if ignore files and folders using tag <ignore> and restrict option (string or sregex) in both options.
- [x] Check if delete content in _/var/ossec/queue/diff_ when deleting any tag <directories report_changes="yes">
- [x] Check if delete content in _/var/ossec/queue/diff_ when report_changes option passed yes to no.
- [x] Check that duplicate entries are ignored. Include entries with a trailing slash in a single comma-separated stanza.

### Linux

- [x] Check syscheck alert for adding a file
- [x] Check syscheck alert for adding text to a file
- [x] Check syscheck alert with report_changes option
- [x] Check syscheck alert for changing owner of file
- [x] Check syscheck alert for changing group of file
- [x] Check syscheck alert for changing file permissions
- [x] Check values different values for options 'check_sum', 'check_md5sum', 'check_sha1sum','check_sha256sum'
- [x] Check syscheck alert for deleting a file
- [x] Check recursion level option (recursion_level=0, recursion_level=2 check folder level 0, 1, 2 and 3)
- [x] Check the option `follow_symbolic_link` (Since v3.8.0)

### Windows

- [x] Check syscheck alert for adding a file
- [x] Check syscheck alert for adding text to a file
- [x] Check syscheck alert for deleting text from a file
- [x] Check syscheck alert with report_changes option and `last-entry` files are stored compressed.
- [x] Check syscheck alert for changing owner of file
- [x] Check syscheck alert for changing file permissions
- [x] Check values diferent values for options 'check_sum', 'check_md5sum', 'check_sha1sum','check_sha256sum'
- [x] Check syscheck alert for deleting a file
- [x] Check recursion level option (recursion_level=0, recursion_level=2 check folder level 0, 1, 2 and 3)

### Linux

- [x] Check syscheck alert for adding a file
- [x] Check syscheck alert for adding text to a file
- [x] Check syscheck alert for deleting text from a file
- [x] Check syscheck alert with report_changes option
- [x] Check the "nodiff" option to don't show the changes of a file
- [x] Check syscheck flag auto_ignore with attributes "frequency" and "timeframe" (1)
- [x] Check syscheck alert for changing owner of file
- [x] Check syscheck alert for changing group of file
- [x] Check syscheck alert for changing file permissions
- [x] Check values diferent values for options 'check_sum', 'check_md5sum', 'check_sha1sum','check_sha256sum'
- [x] Check syscheck alert for deleting a file
- [x] Check recursion level option (recursion_level=0, recursion_level=2 check folder level 0, 1, 2 and 3)
- [x] Check the option `follow_symbolic_link` (Since v3.8.0)

### Windows

- [x] Check syscheck alert for adding a file
- [x] Check syscheck alert for adding text to a file
- [x] Check syscheck alert for deleting text from a file
- [x] Check syscheck alert with report_changes option
- [x] Check the "nodiff" option to don't show the changes of a file
- [x] Check syscheck flag auto_ignore with attributes "frequency" and "timeframe" (1)
- [x] Check syscheck alert for changing attributes of a file
- [x] Check syscheck alert for changing file permissions
- [x] Check values diferent values for options 'check_sum', 'check_md5sum', 'check_sha1sum','check_sha256sum'
- [x] Check syscheck alert for deleting a file
- [x] Check recursion level option (recursion_level=0, recursion_level=2 check folder level 0, 1, 2 and 3)

(1) https://github.com/wazuh/wazuh/pull/605

### Linux

- [x] Check syscheck alert for adding a file
- [x] Check syscheck alert for moving a file
- [x] Check syscheck alert for modifying a file
- [x] Check syscheck alert for deleting a file
- [x] Check syscheck alert for deleting a folder
- [x] Check syscheck alert for changing file permissions
- [x] Check syscheck alert for changing owner of file
- [x] Check syscheck alert for changing group of file
- [x] Check recursive commands.
- [x] Check audit rules added (auditctl -l)
- [x] Check audit rules removed (auditctl -l)
- [x] Remove audit rule manually and check if the rule is reloaded. (Auto-reload each 30 seconds)
- [ ] Remove rules 5 times and check if whodata stops and realtime is started. (Check alert)
- [x] Remove monitored folder and check if the rule is removed and re-add the folder. The rule must be re-added.
- [x] Add blocking rule in audit and check whodata logs and alert. (auditctl -a never,task)
- [x] Restart auditd. Check whodata connection retries.
- [ ] Stop auditd. Move to realtime.
- [ ] Check modifications in a file changed from whodata to realtime.
- [x] Check whodata in Amazon Linux.
- [x] Check the option `follow_symbolic_link` only for directories, not files. (Since v3.8.0)

### Windows

- [x] Check syscheck alert for adding a file
- [ ] Check syscheck alert for moving a file
- [x] Check syscheck alert for modifying a file
- [x] Check syscheck alert for deleting a file
- [x] Verify that whodata does not block files. https://github.com/wazuh/wazuh/pull/3872

**Comment by jesusjimsa** [comment]
When using option `check_md5sum` and restarting the manager, the log still shows all hashes, not only the MD5 one. This happens with the other `check_` options too.

**Comment by jesusjimsa** [comment]
When option `recursion_level` is set to `0`, changes in `/etc/test/test.txt` still appear in the log.

**Comment by jesusjimsa** [comment]
when removing rules 5 times, who data is stopped, but realtime is not started.

**Comment by jesusjimsa** [comment]
On Linux, when stopping `auditd`, realtime is not started.

**Comment by SalvaMilenario** [comment]
Hi 

In the test `Check syscheck alert for moving a file`, we generate two alerts, one to add and others to delete. In the first alert doesn't include the information of Authid.

We have used this configuration:
```xml
  <syscheck>
    
    <disabled>no</disabled>

    <!-- Frequency that syscheck is executed default every 12 hours -->
    <frequency>30</frequency>

    <!-- Default files to be monitored. -->
    <directories check_all="yes" whodata="yes">C:\Users\IEUser\Desktop\test0</directories>

    <!-- Frequency for ACL checking (seconds) -->
    <windows_audit_interval>300</windows_audit_interval>
  </syscheck>
```
This is the alerts that is generated from moving a file:
```
** Alert 1571829749.259121: - ossec,syscheck,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,
2019 Oct 23 11:22:29 (win10-agente5) 172.16.1.21->syscheck
Rule: 554 (level 5) -> 'File added to the system.'
File 'c:\users\ieuser\desktop\test0\new folder\hol.txt' was added.

Attributes:
 - Size: 2
 - Date: Tue Oct 22 15:53:26 2019
 - User: IEUser (S-1-5-21-3859058339-3768143778-240673529-1000)
 - MD5: e47ca7a09cf6781e29634502345930a7
 - SHA1: 0343bb07c98f8a943e8eb80c0ba3d9758d372d22
 - SHA256: a8c23cc814179578e3a774418ac5fc4702a66eb3b78c876df81b290465e6e334
 - File attributes: ARCHIVE
 - Permissions: 
   SYSTEM  (ALLOWED) - DELETE, READ_CONTROL, WRITE_DAC, WRITE_OWNER, SYNCHRONIZE, FILE_READ_DATA, FILE_WRITE_DATA, FILE_APPEND_DATA, FILE_READ_EA, FILE_WRITE_EA, FILE_EXECUTE, FILE_READ_ATTRIBUTES, FILE_WRITE_ATTRIBUTES
   Administrators  (ALLOWED) - DELETE, READ_CONTROL, WRITE_DAC, WRITE_OWNER, SYNCHRONIZE, FILE_READ_DATA, FILE_WRITE_DATA, FILE_APPEND_DATA, FILE_READ_EA, FILE_WRITE_EA, FILE_EXECUTE, FILE_READ_ATTRIBUTES, FILE_WRITE_ATTRIBUTES
   IEUser  (ALLOWED) - DELETE, READ_CONTROL, WRITE_DAC, WRITE_OWNER, SYNCHRONIZE, FILE_READ_DATA, FILE_WRITE_DATA, FILE_APPEND_DATA, FILE_READ_EA, FILE_WRITE_EA, FILE_EXECUTE, FILE_READ_ATTRIBUTES, FILE_WRITE_ATTRIBUTES
   Users  (ALLOWED) - READ_CONTROL, SYNCHRONIZE, FILE_READ_DATA, FILE_READ_EA, FILE_EXECUTE, FILE_READ_ATTRIBUTES
   Everyone  (ALLOWED) - READ_CONTROL, SYNCHRONIZE, FILE_READ_DATA, FILE_READ_EA, FILE_EXECUTE, FILE_READ_ATTRIBUTES

** Alert 1571829749.260669: - ossec,syscheck,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,
2019 Oct 23 11:22:29 (win10-agente5) 172.16.1.21->syscheck
Rule: 553 (level 7) -> 'File deleted.'
File 'c:\users\ieuser\desktop\test0\hol.txt' was deleted.
(Audit) User: 'IEUser (S-1-5-21-3859058339-3768143778-240673529-1000)'
(Audit) Process id: '2804'
(Audit) Process name: 'C:\Windows\explorer.exe'
```
As we can see the first alert doesn't include the information of Audiit.

This problem will be solved in a future version, 3.12.

Regards
